modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/TreeVisitor.java

```diff
@@ -330,44 +331,73 @@ public void visit(BLangConnector connectorNode) {
             SymbolEnv connectorEnv = SymbolEnv.createConnectorEnv(connectorNode, connectorSymbol.scope, symbolEnv);
 
             // TODO: Handle Annotation attachments
-
-            // Cursor position is calculated against the resource parameter scope resolver
-            cursorPositionResolver = ResourceParamScopeResolver.class;
-            connectorNode.params.forEach(param -> this.acceptNode(param, connectorEnv));
-            connectorNode.varDefs.forEach(varDef -> this.acceptNode(varDef, connectorEnv));
-            connectorNode.actions.forEach(action -> this.acceptNode(action, connectorEnv));
+            if (connectorNode.actions.isEmpty() && connectorNode.varDefs.isEmpty()) {
+                this.isCursorWithinBlock(connectorNode.getPosition(), connectorEnv);
+            } else {
+                // Since the connector def does not contains a block statement, we consider the block owner only.
+                // Here it is Connector Definition
+                this.blockOwnerStack.push(connectorNode);
+                connectorNode.varDefs.forEach(varDef -> {
+                    // Cursor position is calculated against the resource parameter scope resolver
```

Comment:
is this comment correct? isn't it using ConnectorScopeResolver?

Reply:
Fixed